### PR TITLE
Fixed line directive ranges when not applying a line directive

### PR DESCRIPTION
**Pull request by TIHan**
Resolves this issue: https://github.com/dotnet/fsharp/issues/7000

Lexing a line directive would consume a newline which would put the LexBuffer in an invalid state, but is adjusted once the line directive was applied. In tooling, we disable applying the line directive which means the LexBuffer would remain in an invalid state. Adding a newline corrects this in cases where we don't apply the line directive.

### CODE REVIEW

**Comment by TIHan** [comment]
> Stupid question time: this doesn't lead to big allocations, right? Since this is just adding a newline to the buffer, not the whole file?

It doesn't add it to the file, just the buffer. We also do this sort of thing in other places in `lex.fsl` so it's normal.

File: src/fsharp/lex.fsl [hunk 1]
```diff
@@ -524,10 +524,14 @@ rule token args skip = parse
...
-
+          else
+              // add a newline when we don't apply a directive since we consumed a newline getting here
+              newline lexbuf
           token args skip lexbuf 
-        else 
-          if not skip then (HASH_LINE (LexCont.Token !args.ifdefStack)) else token args skip lexbuf }
+        else
+          // add a newline when we don't apply a directive since we consumed a newline getting here
+          newline lexbuf
```
Comment (TIHan) [inline src/fsharp/lex.fsl]: I'm actually unsure about putting this here.

File: src/fsharp/lex.fsl [hunk 2]
```diff
@@ -524,10 +524,14 @@ rule token args skip = parse
           // Construct the new position
           if args.applyLineDirectives then 
               lexbuf.EndPos <- pos.ApplyLineDirective((match file with Some f -> fileIndexOfFile f | None -> pos.FileIndex), line)
-
+          else
+              // add a newline when we don't apply a directive since we consumed a newline getting here
+              newline lexbuf
```
Comment (TIHan) [inline src/fsharp/lex.fsl]: This is the line that really fixes the problem.

**Comment by realvictorprm** [comment]
This is great, you see that @smoothdeveloper :)?

File: src/fsharp/lex.fsl (hunk 1 above)
Comment (TIHan) [reply src/fsharp/lex.fsl]: I can confirm this is needed as well.

I tested by turning "skip" off.

**Pull request merged**